### DISCUSSION TITLE
Musical ensemble

### DISCUSSION THREAD

**Discussion by Ivanzyfer** [FakeFishGames/Barotrauma#14872] [Ideas and suggestions]
There are musical instruments in the game, but they all play different boring melodies. If several players want to take a break and play musical instruments together, then it sounds terrible

It would be nice to add, as it was done in another game about ships, but not underwater and with sails - choose a melody from the list and the rest of the musicians could join in playing together on different musical instruments of the same song, sounding like an ensemble

You can also add the ability to learn new songs by buying them from merchants, each faction has its own songs, study only when reputation is achieved, add "know all the songs" to some of the clown's talents, etc.

**Comment by Ivanzyfer** [comment]
You may need to add percussion instruments to make the music sound better.

This will help you not to get bored on the submarine, waiting for the expedition to return

**Comment by Ivanzyfer** [comment]
I will add more description for understanding my suggestion:

Players choose their desired song by pressing '?' while holding an instrument.

After selecting a song, the first player to start using their instrument will become the lead instrument, and begin to play the main melody of the chosen song. After this, any player that begins to use their instrument will join the lead's song as a backing instrument, playing the song with noticeably less notes, no matter which song is selected.

Multiple states change how the shanty is heard, including being under the effects of ethanol which will result in the songs playing out of tune. Other states that alter the song include being underwater

**Comment by jigardi** [comment]
Have my upvote! : D

I've written some pseudocode for this idea when it was mentioned on the baro-discord previously.

First here: https://discord.com/channels/103209684680323072/1149136708571500564

-and then some discussion here: https://discord.com/channels/103209684680323072/1341074805872656538

**Comment by NoahRoopull** [comment]
I made a mod similar to this idea, using jukebox XML for music selection. Syncing songs up (like in Sea of Thieves) would be a lot more work, but my friends and I have fun with it regardless.

https://steamcommunity.com/sharedfiles/filedetails/?id=3433094024
